### PR TITLE
Add a gefs test

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**
This PR:
- temporarily adds a hack to allow testing of the GEFS forecast until the staging is automated.
- adds a test for GEFS
- updates `setup_expt.py` to compartmentalize gfs and gefs setup options.  gefs has a single mode `forecast-only` and is necessary for `create_experiment.py` to work as designed.  To eliminate it would need major work.

**Type of change**
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

**How Has This Been Tested?**
This should be a companion PR to #1772 
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul I see that this is for S2SA, is there a reason that waves are excluded?

**Comment by aerorahul** [comment]
> @aerorahul I see that this is for S2SA, is there a reason that waves are excluded?

No particular reason.  
This is simply a transformation of the instructions that were used to demonstrate that one can run an GEFS ensemble forecast.
I am not sure if wave ensemble ics are available for this date.
Also, the staging is a hack to ensure the gefs ensemble forecast remains intact when forecasts in the gfs system are updated.

File: workflow/setup_expt.py [hunk 1]
```diff
@@ -455,6 +463,9 @@ def input_args(*argv):
                       default=os.path.join(_top, 'parm/config/gefs'))
     gefs.add_argument('--yaml', help='Defaults to substitute from', type=str, required=False,
                       default=os.path.join(_top, 'parm/config/gefs/yaml/defaults.yaml'))
+    gefs.add_argument('--icsdir', help='full path to initial condition directory
```
Comment (TerrenceMcGuinness-NOAA) [inline workflow/setup_expt.py:466]: putting back this switch in this branch for now is appertained, thanks.

**Review by TerrenceMcGuinness-NOAA — APPROVED** [review]

File: workflow/setup_expt.py [hunk 2]
```diff
@@ -224,7 +224,15 @@ def fill_COMROT_forecasts(host, inputs):
...
     Implementation of 'fill_COMROT' for forecast-only mode
     """
-    print('forecast-only mode treats ICs differently and cannot be staged here')
+    if inputs.system in ['gfs']:
+        print('forecast-only mode treats ICs differently and cannot be staged here')
+    elif inputs.system in ['gefs']:  # Temporarily copy ICs from icsdir into COM for      testing
+        print('temporary hack to stage gefs ICs for testing')
+        comrot = os.path.join(inputs.comrot, inputs.pslot)
+        idatestr = datetime_to_YMDH(inputs.idate)
+        current_cycle_dir = f'gefs.{idatestr[:8]}}'
```
Comment (TerrenceMcGuinness-NOAA) [inline workflow/setup_expt.py]:
```suggestion
        current_cycle_dir = f'gefs.{{idatestr[:8]}}'
```
Missed right handed escaped bracket for single format string

File: workflow/setup_expt.py (hunk 1 above)
Comment (TerrenceMcGuinness-NOAA) [inline workflow/setup_expt.py]:
```suggestion
    gefs.add_argument('--icsdir', help='full path to initial condition directory [temporary hack in place for testing]',
```
Missed end quote before line break

File: workflow/setup_expt.py [hunk 3]
```diff
@@ -455,6 +463,9 @@ def input_args(*argv):
                       default=os.path.join(_top, 'parm/config/gefs'))
     gefs.add_argument('--yaml', help='Defaults to substitute from', type=str, required=False,
                       default=os.path.join(_top, 'parm/config/gefs/yaml/defaults.yaml'))
+    gefs.add_argument('--icsdir', help='full path to initial condition directory
+                      [temporary hack in place for testing]',
```
Comment (TerrenceMcGuinness-NOAA) [inline workflow/setup_expt.py]:
```suggestion
```

**Review by TerrenceMcGuinness-NOAA — DISMISSED** [review]

File: workflow/setup_expt.py (hunk 3 above)
Comment (aerorahul) [reply workflow/setup_expt.py]: addressed

File: workflow/setup_expt.py (hunk 1 above)
Comment (aerorahul) [reply workflow/setup_expt.py]: addressed

File: workflow/setup_expt.py (hunk 2 above)
Comment (aerorahul) [reply workflow/setup_expt.py]: addressed

**Comment by JessicaMeixner-NOAA** [comment]
> > @aerorahul I see that this is for S2SA, is there a reason that waves are excluded?
> 
> No particular reason. This is simply a transformation of the instructions that were used to demonstrate that one can run an GEFS ensemble forecast. I am not sure if wave ensemble ics are available for this date. Also, the staging is a hack to ensure the gefs ensemble forecast remains intact when forecasts in the gfs system are updated.

What is the IC date, number of ensembles and resolution - that's something that can be easily fixed.

**Comment by aerorahul** [comment]
> > > @aerorahul I see that this is for S2SA, is there a reason that waves are excluded?
> > 
> > 
> > No particular reason. This is simply a transformation of the instructions that were used to demonstrate that one can run an GEFS ensemble forecast. I am not sure if wave ensemble ics are available for this date. Also, the staging is a hack to ensure the gefs ensemble forecast remains intact when forecasts in the gfs system are updated.
> 
> What is the IC date, number of ensembles and resolution - that's something that can be easily fixed.

See the [ci/cases/C48_S2SA_gefs.yaml](https://github.com/NOAA-EMC/global-workflow/pull/1774/files#diff-87486e534b9c4b514e2c88cd466f252e719e3754681b206adfb6a71f57794b0b) file added in this PR for the initial condition date, ensemble size and resolution.

**Comment by WalterKolczynski-NOAA** [comment]
Yup, the reason I didn't run waves in my GEFS instructions (which Rahul presumably used to make the test) was because we don't have wave ICs (at least not in the official place with the others).

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending CI tests

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**